### DISCUSSION TITLE
unnamed module of loader

### DISCUSSION THREAD

**Discussion by zzhi** [apache/skywalking#7730] [Q&A]
apm-grpc-1.x-plugin-8.7.0.jar  cause an error 

Caused by: java.lang.ClassCastException: class io.grpc.ClientInterceptors$InterceptorChannel cannot be cast to class io.grpc.ManagedChannel (io.grpc.ClientInterceptors$InterceptorChannel and io.grpc.ManagedChannel are in unnamed module of loader org.springframework.boot.loader.LaunchedURLClassLoader @ccb4b1b)

the code: 
`ManagedChannel channe2 = (ManagedChannel) blockingStub.getChannel();`

**Comment by wu-sheng** [comment]
Maybe you are using some specific classloader? Just guess.

**Reply by zzhi** [reply]
before use  java agent plugins  it ok ; after use agent plugins ,throw error like that . when i delete only  the apm-grpc-1.x-plugin-8.7.0.jar  , ok again

**Comment by zzhi** [comment]
after
![521631790695_ pic_hd](https://user-images.githubusercontent.com/10458695/133604675-3e0e71bc-1123-4f04-8a4e-8cc9c5d411ae.jpg)


before 
![531631790799_ pic](https://user-images.githubusercontent.com/10458695/133604706-1b99df65-8c2e-4b98-ac60-8f1863545b4e.jpg)

**Comment by zzhi** [comment]
debug the code 
i find the  org.apache.skywalking.apm.plugin.grpc.v1.client.AbstractStubInterceptor 

change ManagedChannel to io.grpc.ClientInterceptors$InterceptorChannel 
how  to get the ManagedChannelOrphanWrapper from  ClientInterceptors$InterceptorChannel 's channel
@wu-sheng

**Reply by zzhi** [reply]
![image](https://user-images.githubusercontent.com/10458695/133622140-9a26cf19-c66e-4b34-a775-3b6f5e5dbef2.png)

**Reply by wu-sheng** [reply]
Maybe you are in some special cases or unsupported versions, check here about what we are testing. https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/grpc-scenario

These were tested versions, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/grpc-scenario/support-version.list

**Reply by zzhi** [reply]
thanks

**Comment by zyk888333** [comment]
hello, how did you solve this problem?
